mopidy/models.py

```diff
@@ -2,30 +2,156 @@
 
 import json
 
+# TODO: split into base models, serialization and fields?
+
+
+class Field(object):
+    def __init__(self, default=None, type=None, choices=None):
+        """
+        Base field for use in :class:`ImmutableObject`. These fields are
+        responsible type checking and other data sanitation in our models.
+
+        For simplicity fields use the Python descriptor protocol to store the
+        values in the instance dictionary. Also note that fields are mutable if
+        the object they are attached to allow it.
+
+        Default values will be validated with the exception of :class:`None`.
+
+        :param default: default value for field
+        :param type: if set the field value must be of this type
+        :param choices: if set the field value must be one of these
+        """
```

Comment:
Just field or all of them? Also I think this is a candidate for doing a internals vs externals split, as if I was a frontend / js dev I wouldn't care about any of this.